Fix loads the entire frame on the iframe. ⚡

Nishitbaria opened the pull request:
## Fixes Issue
#1612 

Closes #issue_number
#1612 

## Changes proposed

- Added a state variable `showIframe` to track whether the iframe should be rendered or not.
- Modified the `handleButtonClick` function to toggle the value of `showIframe` when the button is clicked.
- Wrapped the `iframe` element with a conditional statement to render it only when `showIframe` is `true`.
- Update



<!-- Describe the changes that you have done in the PR -->

## Screenshots


https://github.com/Spyware007/Animating-Buttons/assets/85815172/6592efbd-5fe3-47c1-a9c1-1a3c45c9679c





## Checklist

<!-- Tick the checkboxes to ensure you've done everything correctly -->

<!--
Example how to mark a checkbox:-
- [x] My code follows the code style of this project.
-->

- [x] You've linked this PR to the correct issue.
- [x] You have checked that the code is working correctly.
- [x] You ⭐️ the repository!

## Note to reviewers

<!-- List anything note-worthy here (potential issues, PR #25 needs to be merged to before working, etc.). -->

Nishitbaria commented:
hey @Spyware007 review PR

adity2911 commented:
I think on click, the iframe disappears?

Nishitbaria commented:
@adity2911  please send me video of your problem because i am unable to see issue of your problem?

adity2911 commented:
You will have to record a video and show it to me. I cannot view buttonsCard on the deployement page.

Nishitbaria commented:
hey @adity2911   You are not able to see button cards because your firebase quota is reached you can check this screenshot,by going to website,On my local I have created buttons and then showing you demo,you can your main website also

adity2911 commented:
![image](https://github.com/Spyware007/Animating-Buttons/assets/72987111/e41f5913-4d85-450c-9d36-4762102a4bfc)
Not working still.

Nishitbaria commented:
check this on your main website @adity2911 


![WhatsApp Image 2023-06-29 at 00 24 58](https://github.com/Spyware007/Animating-Buttons/assets/85815172/881dfd51-3a02-4fbe-97cf-94713463ca5c)

adity2911 commented:
I can use my firebase alright.
I tried to run the code on my local machine and it failed there.

Nishitbaria commented:
@adity2911 Make Sure that you have enable github Login in firebase Authentication Section with 
Client Id and Client Secret 
and also share the video which will be better so we can inspect the issue in best way

adity2911 commented:
It is the same issue as what i had mentioned on my issues tab.
check the video i uploaded there once again. It's the same issue now also.

Nishitbaria commented:
https://animating-buttons-silk.vercel.app/
@adity2911 Kindly check my deployment link I have hosted from vercel and I think you haven't add you domain in firebase Authorized domain section as mentioned below so kindly check upon it

until that you can check my deployment link

![image](https://github.com/Spyware007/Animating-Buttons/assets/85815172/cb013842-cb1a-47f6-a4c3-836a1c313e61)

Nishitbaria commented:
You can also check this video for reference
https://github.com/Spyware007/Animating-Buttons/assets/85815172/2b2d7f8c-8bfe-42a6-bb70-e7bb673b9f69

adity2911 commented:
If you look closely at the video, you might notice that the issue was not for all buttons. There are some particular buttons that are facing this issue. Also as i have said, I tried to run your code on my localHost and it was not working there aslo. Plus the code broke the download functionality also.

Nishitbaria commented:
@adity2911  Bro for running my code you need to setup github auth in firebase and add   your url to authorized domains ,then run it on local and you can check my deployed url it contains buttons and tell me which buttons are creating   ,you can check my deployed url by adding buttons ,there is no problem in my  website   ,please merge it And FIRST OF ALL FIX YOUR MAIN WEBSITE ,because it has firebase quota reached  and no user can now see your buttons.

Souravvmishra commented:
@Nishitbaria Can You Explain The Logic, What Changes You Made In Order To Fix The Bug?

Nishitbaria commented:
> @Nishitbaria Can You Explain The Logic, What Changes You Made In Order To Fix The Bug?


- Added a state variable `showIframe` to track whether the iframe should be rendered or not.
- Modified the `handleButtonClick` function to toggle the value of `showIframe` when the button is clicked.
- Wrapped the `iframe` element with a conditional statement to render it only when `showIframe` is `true`.
- Updated testimonial js

Nishitbaria commented:
> @Nishitbaria Can You Explain The Logic, What Changes You Made In Order To Fix The Bug?


- Added a state variable `showIframe` to track whether the iframe should be rendered or not.
- Modified the `handleButtonClick` function to toggle the value of `showIframe` when the button is clicked.
- Wrapped the `iframe` element with a conditional statement to render it only when `showIframe` is `true`.
- Updated testimonial js

adity2911 commented:
I checked the link you provided.
The download button is now broken.
what did you exactly fix?

Nishitbaria commented:
> I checked the link you provided. The download button is now broken. what did you exactly fix?

wait a minute

Souravvmishra commented:
> > @Nishitbaria Can You Explain The Logic, What Changes You Made In Order To Fix The Bug?
> 
> 
> - Added a state variable `showIframe` to track whether the iframe should be rendered or not.
> - Modified the `handleButtonClick` function to toggle the value of `showIframe` when the button is clicked.
> - Wrapped the `iframe` element with a conditional statement to render it only when `showIframe` is `true`.
> - Updated testimonial js         

How Does This Fix The Issue?

Nishitbaria commented:
> > > @Nishitbaria Can You Explain The Logic, What Changes You Made In Order To Fix The Bug?
> > 
> > 
> > 
> > * Added a state variable `showIframe` to track whether the iframe should be rendered or not.
> > * Modified the `handleButtonClick` function to toggle the value of `showIframe` when the button is clicked.
> > * Wrapped the `iframe` element with a conditional statement to render it only when `showIframe` is `true`.
> > * Updated testimonial js
> 
> How Does This Fix The Issue?

The issue described in video by @adity2911 is now resolved, that what I have fixed you can now.  : https://animating-buttons-silk.vercel.app/

adity2911 commented on src/components/common/Card/Card.jsx:
```diff
@@ -47,20 +48,25 @@ export default function Card({ button }) {
     };
     fetchUser();
   }, [user]);
+  const handleButtonClick = () => {
```
the function is added but not called anywhere.

Nishitbaria replied:
It is not need to be called ,That i have kept just for debuggging

Souravvmishra commented:
@Nishitbaria Can You Explain How Adding A State Variable Fixed The Issue? & if I'm Getting you wrong what else you did that fixed the issue.

Nishitbaria commented:
> @Nishitbaria Can You Explain How Adding A State Variable Fixed The Issue? & if I'm Getting you wrong what else you did that fixed the issue.



> @Nishitbaria Can You Explain How Adding A State Variable Fixed The Issue? & if I'm Getting you wrong what else you did that fixed the issue.

Main thing is that i have solved this issue ,it is working fine as per your requirements ,you can merge my pull request

Souravvmishra commented:
& Just for the reference purpose, even if we remove the changes you have done, it will still work fine for your buttons. You can check it out yourself.

Nishitbaria commented:
> > > @Nishitbaria Can You Explain How Adding A State Variable Fixed The Issue? & if I'm Getting you wrong what else you did that fixed the issue.
> 
> > Main thing is that i have solved this issue ,it is working fine as per your requirements ,you can merge my pull request
> 
> Sorry Brother, We're not satisfied with your Solution yet

what are you requirement,what you want me to do(As I have resolved issue mentioned by @adity2911 )

Nishitbaria commented:
> & Just for the reference purpose, even if we remove the changes you have done, it will still work fine for your buttons. You can check it out yourself.

Issues are meant to be created and assigned when they are present and when somebody is assigned ,he analyzes your code,add his improvements ,but you assigned me issue,i worked on it and now you are not merging .due to your mistake.
Why you assigned this issue me,didn't you checked this when you assign me this issue ,I have worked on it actively after that,and now you are telling ,you dont need my changes ,when a issue is assigned ,it should be checked and then assigned,that your mistake .

`please merge my pull request ,please dont let  our invested time be wasted 🙏`

Nishitbaria commented:
@adity2911 @Souravvmishra

Souravvmishra commented:
@adity2911 We never refused to merge your code. We'll surely do that once we are satisfied with your Solution.

All you have to explain is how is your solution fixing the issue?
What was the purpose of adding the state variable & that conditional rendering? 
How are all these things fixing the issue?

adity2911 commented:
> > & Just for the reference purpose, even if we remove the changes you have done, it will still work fine for your buttons. You can check it out yourself.
> 
> Issues are meant to be created and assigned when they are present and when somebody is assigned ,he analyzes your code,add his improvements ,but you assigned me issue,i worked on it and now you are not merging .due to your mistake. Why you assigned this issue me,didn't you checked this when you assign me this issue ,I have worked on it actively after that,and now you are telling ,you dont need my changes ,when a issue is assigned ,it should be checked and then assigned,that your mistake .
> 
> `please merge my pull request ,please dont let our invested time be wasted 🙏`

Adding two lines of code that doesnot do anything is not considered to be a investment of time. It's a waste of time.

adity2911 commented:
> > @Nishitbaria We never refused to merge your code. We'll surely do that once we are satisfied with your Solution.
> > All you have to explain is how is your solution fixing the issue? What was the purpose of adding the state variable & that conditional rendering? How are all these things fixing the issue?
> 
> I implemented state variable ,which iframe variable holds the state of iframe and doesn't allow it to re-render until handlebuttonclick is called and i have just defines that function ,for your future use if you want to use in any other case ,i can also remove it .My logic was simple that iframe a state variable,preserves the state and does not change in any case. @Souravvmishra

Can you explain when the state of the variable is cahanged? I dont seem to find any instance where the state is changed.

Nishitbaria commented:
> > > & Just for the reference purpose, even if we remove the changes you have done, it will still work fine for your buttons. You can check it out yourself.
> > 
> > 
> > Issues are meant to be created and assigned when they are present and when somebody is assigned ,he analyzes your code,add his improvements ,but you assigned me issue,i worked on it and now you are not merging .due to your mistake. Why you assigned this issue me,didn't you checked this when you assign me this issue ,I have worked on it actively after that,and now you are telling ,you dont need my changes ,when a issue is assigned ,it should be checked and then assigned,that your mistake .
> > `please merge my pull request ,please dont let our invested time be wasted 🙏`
> 
> Adding two lines of code that doesnot do anything is not considered to be a investment of time. It's a waste of time.me

Your issue was resolved in my 2 lines of code ,Why you assigned  me this issue  ,you should have think that as a maintainer ans this is not good ,i have explain my logic in previous #pl 

> > > @Nishitbaria We never refused to merge your code. We'll surely do that once we are satisfied with your Solution.
> > > All you have to explain is how is your solution fixing the issue? What was the purpose of adding the state variable & that conditional rendering? How are all these things fixing the issue?
> > 
> > 
> > I implemented state variable ,which iframe variable holds the state of iframe and doesn't allow it to re-render until handlebuttonclick is called and i have just defines that function ,for your future use if you want to use in any other case ,i can also remove it .My logic was simple that iframe a state variable,preserves the state and does not change in any case. @Souravvmishra
> 
> Can you explain when the state of the variable is cahanged? I dont seem to find any instance where the state is changed.

You issue requires that state of iframe does not change,because if it changes then iframe will be re-rendered.so i have fixed the state variable.

Spyware007 commented:
@Nishitbaria @adity2911 @Souravvmishra 
Please Stop!
@Nishitbaria Sorry for the inconvenience but whatever you had implemented is not a desired solution! 

Closing this PR!

The simplest solution to this is to remove the hyperlink as per my knowledge for those buttons where anchor tag is used instead of button tag!

This can't be solved by making changes in the code, have to change the HTML code from the backend!

Nishitbaria commented:
> @Nishitbaria @adity2911 @Souravvmishra Please Stop! @Nishitbaria Sorry for the inconvenience but whatever you had implemented is not a desired solution!
> 
> Closing this PR!
> 
> The simplest solution to this is to remove the hyperlink as per my knowledge for those buttons where anchor tag is used instead of button tag!
> 
> This can't be solved by making changes in the code, have to change the HTML code from the backend!

Then why you assign issue when it is resolved already ,thats the problem.

adity2911 commented:
I did not assign you. I asked you to look if you can solve it or not.

Nishitbaria commented:
> I did not assign you. I asked you to look if you can solve it or not.

ok
